core/tx_pool.go

```diff
@@ -1023,9 +1023,9 @@ func (pool *TxPool) Has(hash common.Hash) bool {
 	return pool.all.Get(hash) != nil
 }
 
-// removeTx removes a single transaction from the queue, moving all subsequent
+// RemoveTx removes a single transaction from the queue, moving all subsequent
 // transactions back to the future queue.
-func (pool *TxPool) removeTx(hash common.Hash, outofbound bool) {
+func (pool *TxPool) RemoveTx(hash common.Hash, outofbound bool) {
```

Comment:
```suggestion
func (pool *TxPool) RemoveTx(hash common.Hash, outofbound bool) {
    pool.mu.Lock()
    defer pool.mu.Unlock()
```

safe to add locking here

Reply:
Or rather:

```go
func (pool *TxPool) LockAndRemoveTx(hash common.Hash) {
    pool.mu.Lock()
    defer pool.mu.Unlock()
    pool.RemoveTx(hash, true)
}
```